src/tests/enforce-assertSafe-object-key.test.ts

```diff
@@ -40,6 +40,66 @@ const arr = ['value1', 'value2'];
 console.log(arr[0]);
       `,
     },
+    {
+      // Complex template literals with text should be valid
+      code: `
+const obj = { key1: 'value1', key2: 'value2' };
+const id = 'key1';
+console.log(obj[\`prefix_\${id}_suffix\`]);
+      `,
+    },
+    {
+      // Object property access with a variable directly should be valid
+      code: `
+const obj = { key1: 'value1', key2: 'value2' };
+const id = 'key1';
+console.log(obj[id]);
+      `,
+    },
```

Comment:
```suggestion
      // Object property access with a variable directly should be valid
      code: `
const obj = { key1: 'value1', key2: 'value2' };
const id = 'key1';
console.log(obj[id]);
      `,
    },
```
We actually want to make this an invalid case that auto-corrects to:
```
`
import { assertSafe } from 'utils/assertions';

const obj = { key1: 'value1', key2: 'value2' };
const id = 'key1';
console.log(obj[assertSafe(id)]);
      `
```
Please do this for ALL direct object property access.